### PR TITLE
Make internal/testdata and consumer their own go modules

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Part of https://github.com/open-telemetry/opentelemetry-collector/issues/5550.

This makes it possible to use consumer without pulling in the entire base module.  The consumer module depends on internal/testdata

### CODE REVIEW

**Comment by dashpole** [comment]
I chose to add a go module in consumer, rather than consumer/consumererror because they have roughly the same set of dependencies (both depend only on pdata).  I can change this if requested.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -18,6 +18,14 @@ updates:
...
     schedule:
       interval: "weekly"
+  - package-ecosystem: "gomod"
+    directory: "/consumer"
+    schedule:
+      interval: "weekly"
+  - package-ecosystem: "gomod"
+    directory: "/internal/testdata"
+    schedule:
+      interval: "weekly"
```
Comment (bogdandrutu) [inline .github/dependabot.yml:28]: This is a package that we don't want to expose.

File: consumer/consumererror/throttle.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	return "Throttle (" + t.delay.String() + "), error: " + t.err.Error()
+}
+
+func (t throttled) Unwrap() error {
+	return t.err
+}
+
+// ThrottleRetryDelay returns the duration of time we should wait to retry, or
+// zero if the error is not a throttled retry error
+func ThrottleRetryDelay(err error) time.Duration {
```
Comment (bogdandrutu) [inline consumer/consumererror/throttle.go:40]: Why this instead of exposing "Throttle" type and add a func on it `RetryDelay`?

**Review by bogdandrutu — COMMENTED** [review]
I would for sure separate the 2 independent changes:
1. The move of the error.
2. The exposing of the package.

File: .github/dependabot.yml (hunk 1 above)
Comment (dashpole) [reply .github/dependabot.yml:28]: I don't believe this exposes any additional modules.  They are still internal, IIUC.  It does reduce the number of dependencies you pull in when you reference this package.

File: consumer/consumererror/throttle.go (hunk 2 above)
Comment (dashpole) [reply consumer/consumererror/throttle.go]: Moved to #5569

File: .github/dependabot.yml (hunk 1 above)
Comment (dashpole) [reply .github/dependabot.yml:28]: I started working on moving required functionality into the consumer package, but it involves copying basically the entire testdata package into consumer, and didn't seem worth it.  The cleanest way would be just to duplicate the package inside consumer: https://github.com/open-telemetry/opentelemetry-collector/pull/5578

**Pull request closed without merging**